tests/test_fields.py

```diff
@@ -1327,6 +1327,76 @@ def test_timestamp_field_create_like(self):
                 self.assertEqual(0, len(g.data))
 
 
+class TestFieldWhereFunc(unittest.TestCase):
+
+    def test_module_where_numeric(self):
+        input_data = [1,2,3,4]
+        data = np.asarray(input_data, dtype=np.int32)
+        bio = BytesIO()
+        with session.Session() as s:
+            src = s.open_dataset(bio, 'w', 'src')
+            df = src.create_dataframe('df')
+            f = df.create_numeric('foo', 'int32')
+            f.data.write(data)
+
+            r = fields.where(f > 2, 1, 0)
+            self.assertEqual(r.tolist(), [0,0,1,1])
+
+    def test_instance_where_numeric(self):
+        input_data = [1,2,3,4]
+        data = np.asarray(input_data, dtype=np.int32)
+        bio = BytesIO()
+        with session.Session() as s:
+            src = s.open_dataset(bio, 'w', 'src')
+            df = src.create_dataframe('df')
+            f = df.create_numeric('foo', 'int32')
+            f.data.write(data)
+            r = f.where(f > 2, 0)
+            self.assertEqual(r.tolist(), [0,0,3,4])
+
+    def test_instance_where_numeric_inplace(self):
+        input_data = [1,2,3,4]
+        data = np.asarray(input_data, dtype=np.int32)
+        bio = BytesIO()
+        with session.Session() as s:
+            src = s.open_dataset(bio, 'w', 'src')
+            df = src.create_dataframe('df')
+            f = df.create_numeric('foo', 'int32')
+            f.data.write(data)
+
+            r = f.where(f > 2, 0)
+            self.assertEqual(list(f.data[:]), [1,2,3,4])
+            r = f.where(f > 2, 0, inplace=True)
+            self.assertEqual(list(f.data[:]), [0,0,3,4])
+
+    def test_instance_where_with_callable(self):
+        input_data = [10,20,30,40]
+        data = np.asarray(input_data, dtype=np.int32)
+        bio = BytesIO()
+        with session.Session() as s:
+            src = s.open_dataset(bio, 'w', 'src')
```

Comment:
Tests should not only make use of in order data, but should ensure the code is robust to arbitrarily sorted inputs